OpenRA.Mods.Common/Scripting/Global/ReinforcementsGlobal.cs

```diff
@@ -161,12 +161,13 @@ public Actor[] Reinforce(Player owner, string[] actorTypes, CPos[] entryPath, in
 					{
 						var mobiles = cargo != null ? cargo.Passengers.Select(a =>
 						{
-							var mobileInfo = a.Info.TraitInfoOrDefault<MobileInfo>();
```

Comment:
I believe this can simplify down to a much more readable and marginally more efficient:
```csharp
var locomotors = cargo.Passengers
	.Select(a => a.Info.TraitInfo<MobileInfo>())
	.Where(m => m != null)
	.Distinct()
	.Select(m => m.LocomotorInfo)
	.ToList();

foreach (var c in transport.World.Map.FindTilesInCircle(destination, dropRange))
{
	if (!aircraft.CanLand(c))
		continue;

	if (!locomotors.All(m => domainIndex.IsPassable(destination, c, m, (uint)m.GetMovementClass(owner.World.Map.Rules.TileSet))))
		continue;

	destination = c;
	break;
}
```